graylog2-server/src/main/java/org/graylog2/bootstrap/CmdLineTool.java

```diff
@@ -272,11 +271,23 @@ public void run() {
     }
 
     public void doRun(Level logLevel) {
+        final PluginLoaderConfig pluginLoaderConfig = getPluginLoaderConfig(configFile);
+
+        // Move the zstd temp folder from /tmp to our native lib dir to avoid issues with noexec-mounted /tmp directories.
+        // See: https://github.com/Graylog2/graylog2-server/issues/17837
+        // WARNING: This needs to be set before the first use of the zstd library. Our in-memory logger is using
+        //          zstd library, so we need to set it before the first usage of the Logger instance.
+        //          Setting it after the first library usage wouldn't have any effect.
+        if (Native.isLoaded()) {
+            LOG.warn("The zstd library is already loaded. Setting the ZstdTempFolder property doesn't have any effect!");
+        }
+        System.setProperty("ZstdTempFolder", pluginLoaderConfig.getNativeLibDir().toAbsolutePath().toString());
```

Comment:
I think `data/libnative` will usually not exist yet on a new setup. For the first server start, zstd-jni will fail to extract its libraries. When the server is restarted, our netty libs will be present in that directory so that it exists.

Reply:
Good catch, thank you! I adjusted the code to create the native lib dir.